src/Features/Core/Portable/Diagnostics/IDiagnosticAnalyzerService.cs

```diff
@@ -86,6 +86,16 @@ internal interface IDiagnosticAnalyzerService
         /// </para>
         /// </summary>
         Task<ImmutableArray<DiagnosticData>> GetDiagnosticsForSpanAsync(Document document, TextSpan? range, string? diagnosticId = null, bool includeSuppressedDiagnostics = false, CodeActionRequestPriority priority = CodeActionRequestPriority.None, Func<string, IDisposable?>? addOperationScope = null, CancellationToken cancellationToken = default);
```

Comment:
as per above convos, make this an extension.  i prefer the interface to ejust expose the single thing an impl reallyh needs to provide.